packages/multicast_dns/lib/multicast_dns.dart

```diff
@@ -48,14 +48,14 @@ class MDnsClient {
 
   bool _starting = false;
   bool _started = false;
-  RawDatagramSocket _incoming;
   final List<RawDatagramSocket> _sockets = <RawDatagramSocket>[];
   final LookupResolver _resolver = LookupResolver();
   final ResourceRecordCache _cache = ResourceRecordCache();
   final RawDatagramSocketFactory _rawDatagramSocketFactory;
 
-  InternetAddress _mDnsAddress;
-  int _mDnsPort;
+  InternetAddress? _mDnsAddress;
+  late int _mDnsPort;
```

Comment:
The lookup of this value is protected by `_started`, which makes `late` ok, but what about making it nullable and explicitly checking for it before it's used.  I think it has the added benefit that removes the assumption that a separate bit keeps track of whether it's safe to read the value.

Reply:
done